grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -350,6 +376,20 @@ class StockMovementApiController {
         if (packPageItems) {
             createPackPageItemsFromJson(stockMovement, packPageItems)
         }
+
+        if (stockMovement.statusCode == StockMovementStatusCode.DISPATCHED.toString()) {
+            Location inventoryLocation = Location.findByLocationNumber(stockMovement.identifier)
```

Comment:
I still think we want to look this up by both the parent location (stockMovement.destination) and receiving location. And I believe the identifier is only part of the receiving location name. See LocationService.getReceivingLocationName().

Reply:
Also, why did you decide to add the logic to the bindStockMovement() method? It feels like the updateShipment() method would be a better place.

Reply:
Reg 1st comment: After binding, I think I won't see the old destination, hence I was looking by the locationNumber which is unique and is assigned during the creation of an internal location while entering receipt. But sure I'll revisit it and try to look for this location with LocationService.getReceivingLocationName() and LocationService.findInternalLocation().
Reg 2nd comment: I do not recall the particular reason, I think that while I was debugging it I stopped there. I think you're right, the updateShipment will be better.

Reply:
Reg Reg 1st I see that requisition and shipment still have this data.